src/main/java/games/strategy/triplea/delegate/AbstractBattle.java

```diff
@@ -76,14 +76,14 @@
 
   @Override
   public Collection<Unit> getDependentUnits(final Collection<Unit> units) {
-    final Collection<Unit> rVal = new ArrayList<>();
+    final Collection<Unit> dependentUnits = new ArrayList<>();
```

Comment:
Out-of-scope: I believe a HashSet would be the better implementation here, duplicate values are not needed in this case.
Also this could be replaced by a nice stream:
`return units.stream().map(m_dependentUnits::get).filter(Objects::nonNull).flatMap(Collections::stream).collect(Collections.toSet());`

Reply:
I agree that duplicates should probably be removed in the two cases you highlighted.  However, these two methods are called in about 15 locations, and I wouldn't feel comfortable making such a change unless I could be sure removing the duplicates wouldn't cause an unintended side effect at any of those call sites.  This is all buried in combat-related logic, and it seems a bit risky to potentially break such a code hotspot.  Unfortunately, there's not a lot of unit tests in this area of the code. 😞